webui/react/src/pages/F_ExpList/F_ExperimentList.tsx

```diff
@@ -161,95 +172,114 @@ const F_ExperimentList: React.FC<Props> = ({ project }) => {
     }
   }, []);
 
-  useEffect(() => {
-    setSearchParams((params) => {
-      if (page) {
-        params.set('page', page.toString());
-      } else {
-        params.delete('page');
-      }
-      return params;
-    });
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [page, sortString]);
+  const resetPagination = useCallback(() => {
+    setIsLoading(true);
+    setPage(0);
+    setExperiments(INITIAL_LOADING_EXPERIMENTS);
+  }, [setPage]);
 
   useEffect(() => {
-    // useSettings load the default value first, and then load the data from DB
-    // use this useEffect to re-init the correct useSettings value when settings.filterset is changed
-    if (isLoadingSettings) return;
-    const formSetValidation = IOFilterFormSet.decode(JSON.parse(settings.filterset));
-    if (isLeft(formSetValidation)) {
-      handleError(formSetValidation.left, {
-        publicSubject: 'Unable to initialize filterset from settings',
+    let cleanup: () => void;
+    // we only want to init the filterset state from settings once and only when
+    // the settings have finished loading. subscribing to the project settings
+    // observable only works on initial loads -- the settings store will be
+    // fully loaded when navigating to the page in-router, so we check if we
+    // need to subscribe at all.
+    const initFormset = (ps: Loadable<ProjectSettings | null>) => {
+      ps.forEach((s) => {
```

Comment:
unrelated to this pr but `forEach` is a really misleading name for this method. something like `ifLoaded` seems like it would be more clear

Reply:
i agree but see why we made the decision -- it's not unheard of to use part of the array api to describe the methods of a generic monad because lists are monads that developers are familiar with.